binchihua/src/common/list.h

```diff
@@ -168,16 +168,17 @@
 #define LIST_FOREACH_AFTER(name, i, p) for ((i) = (p)->name##_next; (i); (i) = (i)->name##_next)
 
 /* Iterate through all the members of the list p is included in, but skip over p */
-#define LIST_FOREACH_OTHERS(name, i, p)              \
-        for (({                                      \
-                     (i) = (p);                      \
-                     while ((i) && (i)->name##_prev) \
-                             (i) = (i)->name##_prev; \
-                     if ((i) == (p))                 \
-                             (i) = (p)->name##_next; \
-             });                                     \
-             (i);                                    \
-             (i) = (i)->name##_next == (p) ? (p)->name##_next : (i)->name##_next)
+#define LIST_FOREACH_OTHERS(name, i, p)            \
```

Comment:
`LIST_FOREACH_OTHERS` differs from the systemd implementation (on main) and `LIST_POP` was missing, so I copied it from systemd.

Reply:
Nice!